pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java

```diff
@@ -174,6 +174,30 @@ private Object copy(Object value) {
       }
       return list;
     } else if (value.getClass().isArray()) {
+      if (value instanceof byte[]) {
```

Comment:
GenericRow copy logic will fail for primitive array type as they cannot cast to Object[].

Reply:
I don't think we can ingest array of primitive values. See `SegmentColumnarIndexCreator.indexRow()` where MV value has to be `Object[]`.
Which path leads to primitive array getting ingested? Maybe we allow only `byte[]` which is a SV BYTES column?

Reply:
Only keep the check for `byte[]`.